test2code-admin/src/main/kotlin/com/epam/drill/plugins/test2code/SessionHolder.kt

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2020 - 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.drill.plugins.test2code
+
+import com.epam.drill.plugins.test2code.api.*
+import com.epam.drill.plugins.test2code.common.api.*
+import com.epam.drill.plugins.test2code.coverage.*
+import com.epam.drill.plugins.test2code.storage.*
+import com.epam.drill.plugins.test2code.util.*
+import com.epam.dsm.*
+import kotlinx.atomicfu.*
+import kotlinx.collections.immutable.*
+import kotlinx.coroutines.*
+import kotlinx.serialization.*
+import java.lang.ref.*
+
+interface ISessionHolder : Sequence<Session> {
+    val id: String
+    val agentKey: AgentKey
+}
+
+private val logger = logger {}
+
+/**
+ * @param id the sessionHolder ID
+ * @param agentKey todo
+ */
+data class SessionHolder(
+    @Id override val id: String = genUuid(),
+    override val agentKey: AgentKey,
+) : ISessionHolder {
+
+    val sessions = AtomicCache<String, TestSession>()
+
+    override fun iterator(): Iterator<Session> = sessions.values.iterator()
+
+    /**
+     * Start the test session
+     * @features Session starting
+     */
+    fun startSession(
```

Comment:
rename to `createSession` - since we no longer "finish" sessions?

Reply:
renamed